src/datasets/data_files.py

```diff
@@ -823,3 +842,41 @@ def __reduce__(self):
 
         """
         return DataFilesDict, (dict(sorted(self.items())),)
+
+
+def extend_data_files_with_metadata_files_locally(data_files: DataFilesDict, base_path: str) -> None:
+    """
+    Search for metadata files in provided `base_path` for local datasets and add it to provided `data_files` object.
+    """
+    try:
+        metadata_patterns = get_metadata_patterns_locally(base_path)
+    except FileNotFoundError:
+        metadata_patterns = None
+    if metadata_patterns is not None:
+        metadata_files = DataFilesList.from_local_or_remote(metadata_patterns, base_path=base_path)
+        for key in data_files:
+            data_files[key] = DataFilesList(
+                data_files[key] + metadata_files,
+                data_files[key].origin_metadata + metadata_files.origin_metadata,
+            )
```

Comment:
If we implement `DataFilesList.__add__`, I think the logic becomes simple enough not to require a function:

```suggestion
    try:
        metadata_patterns = get_metadata_patterns_locally(base_path)
    except FileNotFoundError:
        metadata_patterns = None
    else:
        metadata_files = DataFilesList.from_local_or_remote(metadata_patterns, base_path=base_path)
        data_files = DatasetFilesDict({key: data_files + metadata_files for key, data_files in data_files.items()})
```